backend/editor/api.py

```diff
@@ -35,4 +57,214 @@ async def pong(response: Response):
+async def findAllStopwords(response: Response):
+    """
+    Get all stopwords within taxonomy
+    """
+    query = """
+        MATCH (n:STOPWORDS) RETURN n
+    """
+    result = session.run(query)
+    allStopwords = [record for record in result]
+    return allStopwords
+
+
+@app.get("/header")
+async def findHeader(response: Response):
+    """
+    Get __header__ within taxonomy
+    """
+    query = """
+        MATCH (n:TEXT) WHERE n.id = "__header__"
+        RETURN n
+    """
+    result = session.run(query)
+    header = [record for record in result]
+    return header
+
+@app.get("/footer")
+async def findFooter(response: Response):
+    """
+    Get __footer__ within taxonomy
+    """
+    query = """
+        MATCH (n:TEXT) WHERE n.id = "__footer__"
+        RETURN n
+    """
+    result = session.run(query)
+    footer = [record for record in result]
+    return footer
+
+# Put methods 
+
+@app.put("/edit/entry/{entry}")
+async def editEntry(request: Request, entry: str):
+    """
+    Editing an entry in a taxonomy.
+    New key-value pairs can be added, old key-value pairs can be updated.
+    URL will be of format '/edit/<id>'
+    """
+    incomingData = await request.json()
+    result = None
+    for key in incomingData.keys():
+        query = f"""
+            MATCH (n:ENTRY) WHERE n.id = $id
+            SET n.{key} = $value
+            RETURN n
+        """
+        result = session.run(query, {"id": entry, "value": incomingData[key]})
+    updatedEntry = [record for record in result]
+    return updatedEntry
+
+@app.put("/edit/synonym/{synonym}")
+async def editSynonyms(request: Request, synonym: str):
+    """
+    Editing a synonym in a taxonomy.
+    New key-value pairs can be added, old key-value pairs can be updated.
+    URL will be of format '/edit/synonym/<id>'
+    """
+    incomingData = await request.json()
+    result = None
+    for key in incomingData.keys():
+        query = f"""
+            MATCH (n:SYNONYMS) WHERE n.id = $id
+            SET n.{key} = $value
+            RETURN n
+        """
+        result = session.run(query, {"id": synonym, "value": incomingData[key]})
+    updatedSynonym = [record for record in result]
+    return updatedSynonym
+
+@app.put("/edit/stopword/{stopword}")
+async def editStopwords(request: Request, stopword: str):
+    """
+    Editing a stopword in a taxonomy.
+    New key-value pairs can be added, old key-value pairs can be updated.
+    URL will be of format '/edit/stopword/<id>'
+    """
+    incomingData = await request.json()
+    result = None
+    for key in incomingData.keys():
+        query = f"""
+            MATCH (n:STOPWORDS) WHERE n.id = $id
+            SET n.{key} = $value
+            RETURN n
+        """
+        result = session.run(query, {"id": stopword, "value": incomingData[key]})
+    updatedStopword = [record for record in result]
+    return updatedStopword
+
+class Header(BaseModel):
+    preceding_lines: List

```

Comment:
put models into their own file.

Reply:
Also I would do a subclass of BaseModel with preceding_lines which would be a parent of Footer and Header.